exchange/exchange.go

```diff
@@ -335,9 +343,9 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 
 		var bidCategory map[string]string
 		//If includebrandcategory is present in ext then CE feature is on.
-		if requestExt.Prebid.Targeting != nil && requestExt.Prebid.Targeting.IncludeBrandCategory != nil {
+		if requestExtLegacy.Prebid.Targeting != nil && requestExtLegacy.Prebid.Targeting.IncludeBrandCategory != nil {
 			var rejections []string
-			bidCategory, adapterBids, rejections, err = applyCategoryMapping(ctx, requestExt, adapterBids, e.categoriesFetcher, targData, &randomDeduplicateBidBooleanGenerator{})
+			bidCategory, adapterBids, rejections, err = applyCategoryMapping(ctx, requestExtLegacy, adapterBids, e.categoriesFetcher, targData, &randomDeduplicateBidBooleanGenerator{})
```

Comment:
From my scan, this is the only `requestExtLegacy` that is not followed by a `.Prebid`. Looking into `applyCategoryMapping()` it only makes use of the `Prebid` component of `requestExtLegacy` I think we can simplify this down to just extracting `requestExtPrebid` from `requestExt` and getting rid of `extractBidRequestExt()` and the extra unmarshal that comes with it.

Reply:
Did you check if `requestExtLegacy.Prebid` usage is purely read-only? I didn't yet check the usage below this line. Currently, `requestExtLegacy ` is a separate copy from the request where a write would be ignored. But if there is no write, then yup - we're safe to make this change.

Reply:
Doesn't matter. `GetPrebid()` returns a copy of the `Prebid` object. Thus `SetPrebid(Prebid)` is required to make the change active, and allows wrapper to set the dirty bit.

Reply:
Please be careful here. It's a shallow copy. Deep pointers can still cause trouble.